.github/workflows/Deploy.yaml

```diff
@@ -3,6 +3,10 @@ name: Deploy
 run-name: "Deploy from [${{ github.ref_name }}] to [${{ inputs.branch }}]"
 
 on:
+  workflow_run:
```

Comment:
Won't this trigger even if I run E2E tests on some other branch? 
![image](https://github.com/microsoft/AL-Go/assets/117829001/126d5c06-a03d-4124-8b9a-f207d6155e81)

Reply:
Yes it will - but the auto-trigger will check that a successful end 2 end test has been run on the latest sha (see checkend2end)

Reply:
Mmh but for example this workflow: 
https://github.com/microsoft/AL-Go/actions/runs/9062344772

The head_sha is 654dd546357a63bd9421137709ad293245095bb2 which is from the main branch but it looks like you ran the E2E test on a different ref (b4407559407171bffd82f47f3064f9c4fae01a7b)? Wouldn't CheckEndToEnd pass in that case?

Reply:
You might be right...
You probably are - will check that.

Reply:
There is no good way of getting the input from an end 2 end test run - so I have added a job to the end 2 end tests which gets skipped if we are running on a specified sha - basically meaning that this end 2 end test isn't useful:
![image](https://github.com/microsoft/AL-Go/assets/10775043/f79f1984-44b7-4788-a874-f1294d3daa33)

This is how it looks when running on an empty ref:
![image](https://github.com/microsoft/AL-Go/assets/10775043/4acf68ad-ad07-4008-8726-f890c2f3801e)

Reply:
Good catch

Reply:
Alternative approach: include the SHA (the input) in the display title of the E2E Tests run.

That way it will be visible which E2E Test run is on _main_ and will save a call to the jobs API.

Reply:
Did a small prototype: https://github.com/mazhelez/BCApps/blob/main/.github/workflows/Test.yaml

![image](https://github.com/microsoft/AL-Go/assets/43066499/8be433c2-8a7d-4882-b18a-02f7490a990a)

Reply:
Good idea - changed the E2E to 
![image](https://github.com/microsoft/AL-Go/assets/10775043/0711d6ce-f1b2-4fae-ac32-617a8b624166)

I still need to check that jobs are not skipped though - else this would be successful:
![image](https://github.com/microsoft/AL-Go/assets/10775043/0a3ab811-3846-46c3-a164-8a426e916f9e)